src/app/import/components/upload/upload.component.ts

```diff
@@ -1,184 +1,162 @@
-
+import { UploadStates } from '../../services/s3.enums';
+import { S3Service } from '../../services/s3.service';
+import { UploadHelper } from '../helpers/upload-helper';
+import { ExistingProjectUploadHelper, LookupListsUploadHelper, RecordAddUploadHelper, RecordNewUploadHelper } from '../helpers/upload-specific';
+import { UploadFileType, UploadOptions, UploadTask, UploadType } from '../helpers/upload-types';
 
 @Component({
   selector: 'kern-upload',
   templateUrl: './upload.component.html',
-  styleUrls: ['./upload.component.scss'],
+  styleUrls: ['./upload.component.scss']
 })
-export class UploadComponent implements OnDestroy, OnInit, AfterViewInit {
-  @Input() projectId: string;
-  @Input() reloadOnFinish: boolean = true;
-  @Input() unique;
-  @Input() init = false;
-  @Input() deleteProjectOnFail = false;
-  file: File | null = null;
+export class UploadComponent implements OnInit {
+
+  @Input() uploadFileType: UploadFileType;
+  @Input() projectId?: string;
+  @Input() uploadOptions: UploadOptions;
+
   @Output() fileAttached = new EventEmitter<File>();
 
+  get UploadFileType(): typeof UploadFileType {
+    return UploadFileType;
+  }
+
   get UploadStatesType(): typeof UploadStates {
     return UploadStates;
   }
-  upload$: Observable<UploadState>;
-  uploadTask$;
-  uploadTaskQuery$;
-  uploadTask;
+
+  @ViewChild('importOptions', { read: ElementRef }) importOptionsHTML: ElementRef;
+  @ViewChild('fileUpload') fileUpload: ElementRef;
+  @ViewChild(LabelStudioAssistantComponent) labelStudioUploadAssistant;
+  uploadHelper: UploadHelper;
+  recordNewUploadHelper: RecordNewUploadHelper;
+  recordAddUploadHelper: RecordAddUploadHelper;
+  existingProjectUploadHelper: ExistingProjectUploadHelper;
+  lookupListsUploadHelper: LookupListsUploadHelper;
+  file: File | null = null;
+  openTab: number = 0;
   uploadStarted: boolean = false;
-  uploadFileType = new FormControl('records');
-  uploadRecordType = new FormControl(RecordCategory.SCALE);
+  uploadTask: UploadTask;
+  uploadTaskQuery$: any;
+  executeOnFinish: () => void;
   uploadType: UploadType = UploadType.DEFAULT;
+  uploadTask$;
+  progressState: UploadState;
+  isProjectTitleEmpty: boolean = false;
+  isProjectTitleDuplicate: boolean = false;
+  submitted: boolean = false;
+  disableInput: boolean = false;
+  tokenizerValuesDisabled: boolean[] = [];
+
+  constructor(private projectApolloService: ProjectApolloService, private router: Router, private s3Service: S3Service) {
+    this.recordNewUploadHelper = new RecordNewUploadHelper(this.projectApolloService, this);
+    this.recordAddUploadHelper = new RecordAddUploadHelper(this);
+    this.existingProjectUploadHelper = new ExistingProjectUploadHelper(this.projectApolloService, this);
+    this.lookupListsUploadHelper = new LookupListsUploadHelper(this)
+    this.uploadHelper = new UploadHelper(this, this.recordNewUploadHelper, this.recordAddUploadHelper, this.existingProjectUploadHelper, this.lookupListsUploadHelper);
```

Comment:
as mentioned above, this procedure seems wrong -> the uploadHelper class is created 5 times in these lines.

Also why do we need to create all of them even if only one is correct for the provided upload type?